protocols/imap/src/main/java/org/apache/james/imap/api/process/ImapSession.java

```diff
@@ -38,40 +36,41 @@
  */
 public interface ImapSession {
     class SessionId {
-        public static SessionId random() {
-            return new SessionId(UUID.randomUUID());
+        private static final RandomStringGenerator RANDOM_STRING_GENERATOR = new RandomStringGenerator.Builder().withinRange('a', 'z').build();
+        private static final int LENGTH = 12;
+
+        public static SessionId generate() {
+            return new SessionId("SID-" + RANDOM_STRING_GENERATOR.generate(LENGTH));
         }
 
-        private final UUID uuid;
+        private final String value;
 
-        private SessionId(UUID uuid) {
-            this.uuid = uuid;
+        private SessionId(String value) {
+            this.value = value;
         }
 
         public String asString() {
-            return uuid.toString();
+            return value;
         }
 
         @Override
         public final boolean equals(Object o) {
             if (o instanceof SessionId) {
                 SessionId sessionId = (SessionId) o;
 
-                return Objects.equals(this.uuid, sessionId.uuid);
+                return Objects.equals(this.value, sessionId.value);
             }
             return false;
         }
 
         @Override
         public final int hashCode() {
-            return Objects.hash(uuid);
+            return Objects.hash(value);
         }
 
         @Override
         public String toString() {
-            return MoreObjects.toStringHelper(this)
-                .add("uuid", uuid)
-                .toString();
+            return asString();
```

Comment:
should be different, no?
for example for a session ID whose value is `SID-fkzeofkpzoe`, asString should be `SID-fkzeofkpzoe` while toString should be `{value=SID-fkzeofkpzoe}`, no?

Reply:
@mbaechler claimed the opposite. Link https://github.com/linagora/james-project/pull/3402#discussion_r432375436

Honestly, I don't care, both are ok to me. I'll keep the current version unless some strong divergences are expressed.

Reply:
I see, it's outside the scope of this PR, so keep it like this for now, we we'll discuss this coding style question somewhere else.